src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,181 @@
+                (int) Utility.minsToSecs(DEFAULT_REFRESH_PERIOD_IN_MINS),
+                UI.status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
+        this.refreshTimer.start();
+        UI.events.registerEvent((RefreshTimerTriggeredEventHandler) this::updateSyncRefreshRate);
+    }
+
+    /**
+     * @return The duration between each refresh of the data store.
+     */
+    public long getRefreshDurationInMinutes() {
+        return refreshDurationInMinutes;
+    }
+
+    /**
+     * Restarts the refreshTimer.
+     */
+    public void restartRefreshTimer(){
+        this.refreshTimer.restart();
+    }
+
+    /**
+     * Updates the period of the refreshTimer for synchronization of the data store.
+     * The initial call to this method does not update the period of the refreshTimer.
+     * @param e RefreshTimerTriggeredEvent object that holds the current API quota information.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        ApiQuotaInfo info = e.getApiQuotaInfo();
+
+        if (remainingApiRequests == UNINITIALIZED){
+            remainingApiRequests = info.getRemainingRequests();
+            return;
+        }
+
+        apiCallsUsedInPreviousRefresh = computeApiCallsUsedInPreviousRefresh(info.getRemainingRequests());
+        remainingApiRequests = info.getRemainingRequests();
+        refreshDurationInMinutes = computeRefreshTimerPeriod(info.getRemainingRequests(),
+                                                             Utility.minutesFromNow(info.getNextRefreshInMillisecs()),
+                                                             apiCallsUsedInPreviousRefresh, API_QUOTA_BUFFER,
+                                                             DEFAULT_REFRESH_PERIOD_IN_MINS);
+        this.refreshTimer.restartTimer((int) Utility.minsToSecs(refreshDurationInMinutes));
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
+                + "mins with API calls used in previous refresh cycle is " + apiCallsUsedInPreviousRefresh
+                + ", current API quota is " + info.getRemainingRequests() + " and next API quota top-up in "
+                + info.getNextRefreshInMinutesFromNow() + "mins.");
+    }
+
+    /**
+     * Computes the API calls used in previous refresh.
+     * @param newRemainingApiRequests The number of API requests remaining in the current API quota window.
+     * @return The number of API calls used in previous refresh or apiCallsUsedInPreviousRefresh.
+     *         apiCallsUsedInPreviousRefresh will be returned when newRemainingApiRequests > remainingApiRequests.
+     */
+    private int computeApiCallsUsedInPreviousRefresh(int newRemainingApiRequests) {
+        int apiUsageInRecentRefresh = remainingApiRequests - newRemainingApiRequests;
+
+        if (apiUsageInRecentRefresh < 0) {
+            return apiCallsUsedInPreviousRefresh;
+        }
+
+        return apiUsageInRecentRefresh;
```

Comment:
How about this?
```java
return remainingApiRequests < newRemainingApiRequests  // A top up has occurred since the previous refresh
        ? newRemainingApiRequests 
        : remainingApiRequests - newRemainingApiRequests;
```

Now that the method has shrunk to one statement, may be we don't need this method anymore?